api-reference/10 UI Components/dxDataGrid/1 Configuration/summary/calculateCustomSummary.md

```diff
@@ -123,6 +121,202 @@ A summary value calculation is conducted in three stages: *start* - the **totalV
+            <DxColumn 
+                data-field="Area" 
+                :calculate-cell-value="calculateArea" 
+            />
+            <DxSummary :calculate-custom-summary="calculateAreaSummary">
+                <DxTotalItem
+                    name="AreaSummary"
+                    summary-type="custom"
+                    show-in-column="Area"
+                    display-format="Total Area: {0}"
+                />
+            </DxSummary>
+        </DxDataGrid>
+    </template>
+    <script>
+    import { DxDataGrid, DxColumn, DxSummary, DxTotalItem } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn,
+            DxSummary, 
+            DxTotalItem
+        },
+        data() {
+            return {
+                // ...
+                calculateArea(rowData) {
+                    return rowData.width * rowData.height;
+                },
+                calculateAreaSummary(options) {
+                    if (options.name === "calculateAreaSummary") {
+                        if (options.summaryProcess === "start") {
+                            options.totalValue = 0;
+                        }
+                        if (options.summaryProcess === "calculate") {
+                            options.totalValue += this.calculateArea(options.value);
+                        }
+                    }
+                },
+            };
+        },
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import DataGrid, { Column, Summary, TotalItem } from 'devextreme-react/data-grid';
+
+    function calculateArea(rowData) {
+        return rowData.width * rowData.height;
+    };
+
+    function calculateAreaSummary(options) {
+        if (options.name === "calculateAreaSummary") {
+            if (options.summaryProcess === "start") {
+                options.totalValue = 0;
+            }
+            if (options.summaryProcess === "calculate") {
+                options.totalValue += calculateArea(options.value);
+            }
+        }
+    }
+
+    function App() {
+        // ...
+        return (
+            <DataGrid ...>
+                <Column dataField="width" />
+                <Column dataField="height" />
+                <Column dataField="Area" calculateCellValue={calculateArea} />
+                <Summary calculateCustomSummary={calculateAreaSummary}>
+                    <TotalItem
+                        name="AreaSummary"
+                        summaryType="custom"
+                        showInColumn="Area"
+                        displayFormat="Total Area: {0}"
+                    />
+                </Summary>
+            </DataGrid>
+        );
+    }
+
+    export default App;
+
+---
+
+Note that the example above uses totalItems.**showInColumn** instead of totalItems.**column**. This is because **calculateCustomSummary** works with values from multiple columns.
```

Comment:
The effect and the cause are switched places. We need entire data objects within calculateCustomSummary. To achieve that, we don't specify the column. Not the other way around. Give this information above the example.